### PR TITLE
Generation of new HA config for A2HA inside automate <#5401>

### PR DESCRIPTION
**Pull request by jayvikramsharma1**
### :nut_and_bolt: Description: What code changed, and why?
As per requirement we need to generate a wrapper config file for automate HA mode deployment
It contains two type of HA mode deployment either AWS or Existing Nodes, base on argument with init-config-ha command

- **chef-automate init-config-ha existing_node**
- **chef-automate init-config-ha aws**

### :chains: Related Resources
https://app.zenhub.com/workspaces/the-delta-quadrant-5fc7c4c3922dc10021323fa6/issues/chef/automate/5401
### :+1: Definition of Done
Written code for generating config.toml file as a first phase, Unit Test cases covered, devevloper testing of acceptance criteria and working behaviour.
### :athletic_shoe: How to Build and Test the Change

- **rebuild components/automate-cli/**
 
### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

**Review by punitmundra — APPROVED** [review]
changes looks to fine.

**Review by karunesh-tech — APPROVED** [review]

**Review by kalroy — CHANGES_REQUESTED** [review]

File: components/automate-cli/cmd/chef-automate/init_config_ha.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+func runInitConfigHACmd(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		msg := "one argument expected as any of deplyment mode like aws or existing_node. "
+		writer.Printf(msg)
+		return status.Wrap(errors.New(msg), status.ConfigError, msg)
+	} else if args[0] == "aws" {
+		writer.Printf("Generating initial automate high availability configuration for AWS deployment")
+		return runInitConfigAwsHACmd()
+	} else if args[0] == "existing_node" {
```
[the remark this answers is not in the document]
Comment (punitmundra) [reply components/automate-cli/cmd/chef-automate/init_config_ha.go:44]: bare-metal

**Comment by jayvikramsharma1** [comment]
@kalroy @punitmundra Please review my PR, I have done review changes

**Review by kalroy — APPROVED** [review]

**Comment by kalroy** [comment]
Code changes look good. But please fix the pipeline before merging.

**Pull request merged**